### PR TITLE
Added unit tests for CampaignSummaryQueryHandler #863

### PR DESCRIPTION
**Pull request by nedruk**
Added unit tests for CampaignSummaryQueryHandler.

### CODE REVIEW

**Comment by nedruk** [comment]
Please review and let me know if that's not sufficient.

Note: the code throws exception if ManagingOrganization returned in the snippet below is null. 

image

Is it something that needs to be fixed?

Thanks

**Comment by mgmccarthy** [comment]
@tonysurma, do you need a code review for this?

**Comment by tonysurma** [comment]
@mgmccarthy if you could please

File: AllReadyApp/Web-App/AllReady.UnitTest/Areas/Admin/Features/Campaigns/CampaignSummaryQueryHandlerTests.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+using Shouldly;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace AllReady.UnitTest.Areas.Admin.Features.Campaigns
+{
+    public class CampaignSummaryQueryHandlerShould : InMemoryContextTest
```
Comment (mgmccarthy) [inline AllReadyApp/Web-App/AllReady.UnitTest/Areas/Admin/Features/Campaigns/CampaignSummaryQueryHandlerTests.cs]: you've changed the name of the class to be correct, but the name of the .cs file in disk still says `CampaignSummaryQueryHandlerTests.cs`. 
image
please update it to the correct name.

File: AllReadyApp/Web-App/AllReady.UnitTest/Areas/Admin/Features/Campaigns/CampaignSummaryQueryHandlerTests.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        }
+
+        [Fact]
+        public async Task ReturnCampaignSummaryViewModel_WhenCampaignExists()
+        {
+            var handler = new CampaignSummaryQueryHandler(this.Context);
+
+            var result = await handler.Handle(new CampaignSummaryQuery { CampaignId = 111 });
+
+            result.ShouldNotBeNull();
```
Comment (mgmccarthy) [inline AllReadyApp/Web-App/AllReady.UnitTest/Areas/Admin/Features/Campaigns/CampaignSummaryQueryHandlerTests.cs]: this assertions does not assert what the test name says it should. The test name is `ReturnCampaignSummaryViewModel_WhenCampaignExists`, yet you're asserting that the result ShouldNotBeNull. You can change this test to assert it returns that specific view model by adding:
`Assert.IsType<CampaignSummaryViewModel>(result);`

File: AllReadyApp/Web-App/AllReady.UnitTest/Areas/Admin/Features/Campaigns/CampaignSummaryQueryHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        }
+
+        [Fact]
+        public async Task SetPrimaryContactInCampaignSummaryViewModel()
+        {
+            Context.Campaigns.First().CampaignContacts = new List<CampaignContact> {
```
Comment (mgmccarthy) [inline AllReadyApp/Web-App/AllReady.UnitTest/Areas/Admin/Features/Campaigns/CampaignSummaryQueryHandlerTests.cs]: wrap the `{` to the next line

File: AllReadyApp/Web-App/AllReady.UnitTest/Areas/Admin/Features/Campaigns/CampaignSummaryQueryHandlerTests.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
+﻿using AllReady.Areas.Admin.Features.Campaigns;
+using AllReady.Models;
+using Shouldly;
+using System;
```
Comment (mgmccarthy) [inline AllReadyApp/Web-App/AllReady.UnitTest/Areas/Admin/Features/Campaigns/CampaignSummaryQueryHandlerTests.cs]: this using is not being used... please delete it

File: AllReadyApp/Web-App/AllReady.UnitTest/Areas/Admin/Features/Campaigns/CampaignSummaryQueryHandlerTests.cs [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+            Context.SaveChanges();
+
+            var handler = new CampaignSummaryQueryHandler(this.Context);
+
+            var result = await handler.Handle(new CampaignSummaryQuery { CampaignId = 111 });
+
+            result.PrimaryContactFirstName.ShouldNotBeNullOrEmpty();
+            result.PrimaryContactLastName.ShouldNotBeNullOrEmpty();
+            result.PrimaryContactEmail.ShouldNotBeNullOrEmpty();
+            result.PrimaryContactPhoneNumber.ShouldNotBeNullOrEmpty();
```
Comment (mgmccarthy) [inline AllReadyApp/Web-App/AllReady.UnitTest/Areas/Admin/Features/Campaigns/CampaignSummaryQueryHandlerTests.cs]: you should test that these four properties on the returned ViewModel are equal to the values you set them to when you create a new Contact in your setup.

**Comment by nedruk** [comment]
Thanks for the review, @mgmccarthy. 
I've made changes as per your recommendations. Should be all good now.

**Comment by mgmccarthy** [comment]
@nedruk, nice job. Thanks for the contribution. @tonysurma :shipit:

**Pull request merged**